x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,574 @@
+----
++
+This command generates a `.zip` file that contains certificates and keys
+to use with {es} and {kib}. Each folder contains a `README.txt`
+explaining how to use these files.
+
+   a. When asked if you want to generate a CSR, enter `n`.
+
+   b. When asked if you want to use an existing CA, enter `y`.
+
+   c. Enter the absolute path to your CA, such as the path to the
+    `elastic-stack-ca.p12` file.
+
+   d. Enter the password for your CA.
+
+   e. Enter an expiration value for your certificate. You can enter the
+   validity period in years, months, or days. For example, enter `90D` for 90
+   days.
+
+   f. When asked if you want to generate one certificate per node, enter `y`.
++
+Each certificate will have its own private key, and will be issued for a
+specific hostname or IP address.
+
+   g. When prompted, enter the name of the first node in your cluster. Use the
+   same node name as the value for the `node.name` parameter in the
+   `elasticsearch.yml` file.
+
+   h. Enter all hostnames used to connect to your first node. These hostnames
+   will be added as DNS names in the Subject Alternative Name (SAN) field in your certificate.
++
+List every hostname and variant used to connect to your cluster over HTTPS.
+
+   i. Enter the IP addresses that clients can use to connect to your node.
+
+   j. Repeat these steps for each additional node in your cluster.
+
+. After generating a certificate for each of your nodes, enter a password for
+   your private key when prompted.
+
+. Unzip the generated `elasticsearch-ssl-http.zip` file. This compressed file
+   contains one directory for both {es} and {kib}.
++
+--
+[source,txt]
+----
+/elasticsearch
+|_ README.txt
+|_ http.p12
+|_ sample-elasticsearch.yml
+----
+
+[source,txt]
+----
+/kibana
+|_ README.txt
+|_ elasticsearch-ca.pem
+|_ sample-kibana.yml
+----
+--
+
+. If necessary, rename the `http.p12` file to match the name of your existing
+certificate for HTTP client communications. This keystore contains both a
+certificate and key.
+
+. Replace your existing keystore (such as `http.p12`) with the new keystore,
+ensuring that the file names match.
++
+IMPORTANT: If your keystore is password protected, the password
+is stored in the {es} keystore, _and_ the password needs to change, then you
+must update the `ES_PATH_CONF/elasticsearch.yml` file to use the filename of the
+new keystore file that you created.
+
+. Add the password for your private key to the secure settings in {es}.
++
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.http.ssl.keystore.secure_password
+----
+
+. Restart the node where you updated the keystore. {es} will reload the file and
+its configuration and start presenting the new  certificate to the incoming TLS
+connections.
+
+. (Optional) Use the <<security-api-ssl,SSL certificate API>> to verify that
+{es} loaded the new keystore.
+
+. On each additional node in your cluster, stop {es} and {kib}, update the
```

Comment:
maybe qualify "update the keystore" more. As in that they have to copy the keystore from the first node where they created all of them, to the specific node and follow a subset of the steps (even better explicitly note down which steps)

Reply:
I'll provide more clarity around these steps.